api/app/auto_spatial_advisory/fuel_stats.py

```diff
@@ -0,0 +1,64 @@
+"""
+Functions for computing fuel type stats
+"""
+from datetime import date
+from typing import List, Optional
+from wps_shared.db.models.auto_spatial_advisory import HfiClassificationThreshold, SFMSFuelType as DBSFMSFuelType
+from wps_shared.schemas.fba import (
+    AdvisoryCriticalHours,
+    ClassifiedHfiThresholdFuelTypeArea,
+    HfiThreshold,
+    SFMSFuelType
+)
+
+"""
+    <!-- Percent Curing default values from June 2021 -->
+    <!-- November 1 to July 15 - 60% -->
+    <!-- July 16 to July 30 - 70% -->
+    <!-- July 31 to August 15 - 80% -->
+    <!-- August 16 - October 31 (end of season) - 90% -->
+"""
+def get_optional_percent_curing(grass_curing_date: date, sfms_fuel_type: DBSFMSFuelType):
+    # 12 is the sfms fuel id for grass types
+    if sfms_fuel_type is None or sfms_fuel_type.fuel_type_id != 12:
+        return None
+    
+    # we can straddle a prev calendar year and the next calendar year with this date range case, nov 1, 2024 and jul 15, 2025
+    if grass_curing_date >= date(grass_curing_date.year, 11, 1) and grass_curing_date <= date(grass_curing_date.year + 1, 7, 15):
```

Comment:
Do we need to be concerned about the year wrapping? If not, it might be easier to understand if the second condition was for <= Dec 31. Though this does work as is, it's just slightly less understandable.

Reply:
I think we do since lots of months/days are <= Dec 31 but are greater than it if you consider the year.

Reply:
Nevermind, captured your further simplification here: https://github.com/bcgov/wps/pull/4344/commits/5f5fa94a75bbfc7ed80dee4402d19539f47dd571